### PR TITLE
Fix broken uiuc.edu links

### PR DESCRIPTION
**Pull request by d-torrance**
We also switch from `http` to `https` for a few macaulay2.com links while we're at it.

Closes: #3585

### CODE REVIEW

File: M2/CMakeLists.txt [hunk 1]
```diff
@@ -58,7 +58,7 @@ file(STRINGS VERSION VERSION)
 project(Macaulay2
   VERSION	${VERSION}
   DESCRIPTION	"A software system for algebraic geometry research"
-  HOMEPAGE_URL	http://macaulay2.com # contact: Macaulay2@math.uiuc.edu
+  HOMEPAGE_URL	https://macaulay2.com # contact: Macaulay2@math.uiuc.edu
```
Comment (mahrud) [inline M2/CMakeLists.txt:61]: The email also needs to change, or just be removed, but not very urgent.

File: M2/Macaulay2/packages/Macaulay2Doc/changes.m2 [hunk 2]
```diff
@@ -2447,7 +2447,7 @@ document {
 			      TO "Text :: Text"
 			      }
 			 },
-		    LI { "downloadable ", HREF{"http://www.math.uiuc.edu/Macaulay2/Packages/", "packages"}, ":",
+		    LI { "downloadable packages:",
 			 UL {
 			      LI { EM "Kronecker", ", Kronecker normal form of a matrix pencil, by Edward Carter" },
 			      LI { EM "LDL", ", the ", TT "LDL'", " factorization of a positive semidefinite matrix, by Helfried Peyrl" }
```
Comment (mahrud) [inline M2/Macaulay2/packages/Macaulay2Doc/changes.m2:2453]: Huh never heard of this package!

File: M2/Macaulay2/packages/Macaulay2Doc/overview3.m2 [hunk 3]
```diff
@@ -679,7 +679,7 @@ document {
      Key => "Other sources of information about Macaulay2",
```
Comment (mahrud) [inline M2/Macaulay2/packages/Macaulay2Doc/overview3.m2:679]: Also never noticed this strange node. Macaulay2Doc is such a maze ...

File: M2/Macaulay2/packages/NumericalSchubertCalculus/README [hunk 4]
```diff
@@ -1,5 +1,3 @@
-http://www.math.uiuc.edu/Macaulay2/Packages/
-
 This is the auxiliary directory for NumericalSchubertCalculus package.
 
 https://github.com/NumericalSchubertCalculus
```
Comment (mahrud) [inline M2/Macaulay2/packages/NumericalSchubertCalculus/README:3]: @antonleykin did you make this github user?

File: M2/libraries/cddlib/Makefile.in [hunk 5]
```diff
@@ -7,7 +7,7 @@ VERSION = 0.94m
 # cddlib is now available on github, so we could switch this library to a submodule
 # https://github.com/cddlib/cddlib
 # https://www.inf.ethz.ch/personal/fukudak/cdd_home/
-URL = http://macaulay2.com/Downloads/OtherSourceCode
+URL = https://macaulay2.com/Downloads/OtherSourceCode
```
Comment (mahrud) [inline M2/libraries/cddlib/Makefile.in:10]: all of these standard URL variables are unnecessary, right?

File: M2/Macaulay2/packages/Cremona/documentation.m2 [hunk 6]
```diff
@@ -332,7 +332,7 @@ Inputs => {
 Ideal => "I" => {"a homogeneous ideal defining a smooth projective variety ",TEX///$X\subset\mathbb{P}^n$///}}, 
 Outputs => { 
 ZZ => {"the topological Euler characteristics of ",TEX///$X$///,"."}}, 
-PARA{"This is an application of the method ", TO "SegreClass",". See also the corresponding methods in the packages ", HREF{"http://www.math.fsu.edu/~aluffi/CSM/CSM.html", "CSM-A"},", by P. Aluffi, and ", HREF{"http://www.math.uiuc.edu/Macaulay2/doc/Macaulay2-1.16/share/doc/Macaulay2/CharacteristicClasses/html/", "CharacteristicClasses"},", by M. Helmer and C. Jost."},
+PARA{"This is an application of the method ", TO "SegreClass",". See also the corresponding methods in the packages ", HREF{"http://www.math.fsu.edu/~aluffi/CSM/CSM.html", "CSM-A"},", by P. Aluffi, and ", TO "CharacteristicClasses::CharacteristicClasses", by M. Helmer and C. Jost."},
```
Comment (mahrud) [inline M2/Macaulay2/packages/Cremona/documentation.m2:335]:
```suggestion
PARA{"This is an application of the method ", TO "SegreClass",". See also the corresponding methods in the packages ", HREF{"http://www.math.fsu.edu/~aluffi/CSM/CSM.html", "CSM-A"},", by P. Aluffi, and ", TO "CharacteristicClasses::CharacteristicClasses", ", by M. Helmer and C. Jost."},
```

**Review by mahrud — APPROVED** [review]

File: M2/Macaulay2/packages/Cremona/documentation.m2 (hunk 6 above)
Comment (d-torrance) [reply M2/Macaulay2/packages/Cremona/documentation.m2:335]: Oops -- thanks!

File: M2/libraries/cddlib/Makefile.in (hunk 5 above)
Comment (d-torrance) [reply M2/libraries/cddlib/Makefile.in:10]: The autotools build uses it to fetch tarballs.

File: M2/CMakeLists.txt (hunk 1 above)
Comment (d-torrance) [reply M2/CMakeLists.txt:61]: Easy enough -- I added another commit fixing it, plus one other location I found with this old address.

File: M2/libraries/cddlib/Makefile.in (hunk 5 above)
Comment (mahrud) [reply M2/libraries/cddlib/Makefile.in:10]: Oh I thought the default one was included from Makefile.library.in. Then it wouldn't need to be repeated unless the url is different.

Comment (d-torrance) [reply M2/libraries/cddlib/Makefile.in:10]: Oh good call!  That definitely would work, but we're not currently doing it.  I'll push another commit with that change

File: M2/Macaulay2/packages/Macaulay2Doc/overview3.m2 [hunk 7]
```diff
@@ -596,7 +596,7 @@ document {
      Key => "The authors",
      PARA{},
      "To communicate with the authors about the program, use this email address:
-     ", HREF {"mailto:Macaulay2@math.uiuc.edu", "<Macaulay2@math.uiuc.edu>"}, ".",
+     ", HREF {"mailto:macaulay2@googlegroups.com", "<macaulay2@googlegroups.com>"}, ".",
```
Comment (mahrud) [inline M2/Macaulay2/packages/Macaulay2Doc/overview3.m2:599]: This could be confusing if someone emails the whole group, no?

**Pull request merged**